dask_cuda/tests/test_worker_spec.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+from distributed import Nanny
+
+from dask_cuda.worker_spec import worker_spec
+
+
+def _check_option(spec, k, v):
+    assert all([s["options"][k] == v for s in spec.values()])
+
+def _check_env_key(spec, k, enable):
+    if enable:
+        assert all([k in s["options"]["env"] for s in spec.values()])
+    else:
+        assert all([k not in s["options"]["env"] for s in spec.values()])
+
+def _check_env_value(spec, k, v):
+    if not isinstance(v, list):
+        v = [v]
+
+    for i in v:
+        assert all([i in set(s["options"]["env"][k].split(",")) for s in spec.values()])
+
+@pytest.mark.parametrize("num_devices", [1, 4])
+@pytest.mark.parametrize("cls", [Nanny])
+@pytest.mark.parametrize("interface", [None, "eth0", "enp1s0f0"])
+@pytest.mark.parametrize("protocol", [None, "tcp", "ucx"])
+@pytest.mark.parametrize("dashboard_address", [None, ":0", ":8787"])
+@pytest.mark.parametrize("threads_per_worker", [1, 8])
+@pytest.mark.parametrize("silence_logs", [False, True])
+@pytest.mark.parametrize("enable_infiniband", [False, True])
+@pytest.mark.parametrize("enable_nvlink", [False, True])
```

Comment:
This is around 1000 tests total.  Maybe this is too much?

Reply:
You're right in number, but it runs in 3 seconds, so maybe it's ok for now?